parzip 1.3.0 fails to build with gcc 13

gcc -v
gcc version 13.0.1 20230127 (Red Hat 13.0.1-0) (GCC)


parzip 1.3.0

  User defined options
    auto_features : enabled
    bindir        : /usr/bin
    buildtype     : plain
    datadir       : /usr/share
    includedir    : /usr/include
    infodir       : /usr/share/info
    libdir        : /usr/lib64
    libexecdir    : /usr/libexec
    localedir     : /usr/share/locale
    localstatedir : /var
    mandir        : /usr/share/man
    prefix        : /usr
    sbindir       : /usr/sbin
    sharedstatedir: /var/lib
    sysconfdir    : /etc
    wrap_mode     : nodownload

Found ninja-1.11.1 at /usr/bin/ninja
+ /usr/bin/meson compile -C redhat-linux-build -j 6 --verbose
ninja: Entering directory `/builddir/build/BUILD/parzip-1.3.0/redhat-linux-build'
[1/18] g++ -Isrc/libparzcore.a.p -Isrc -I../src -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -std=c++17 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fPIC -pthread -MD -MQ src/libparzcore.a.p/utils.cpp.o -MF src/libparzcore.a.p/utils.cpp.o.d -o src/libparzcore.a.p/utils.cpp.o -c ../src/utils.cpp
FAILED: src/libparzcore.a.p/utils.cpp.o
g++ -Isrc/libparzcore.a.p -Isrc -I../src -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -std=c++17 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fPIC -pthread -MD -MQ src/libparzcore.a.p/utils.cpp.o -MF src/libparzcore.a.p/utils.cpp.o.d -o src/libparzcore.a.p/utils.cpp.o -c ../src/utils.cpp
In file included from ../src/utils.h:20,
                 from ../src/utils.cpp:18:
../src/file.h:56:5: error: ‘uint64_t’ does not name a type
   56 |     uint64_t size() const;
      |     ^~~~~~~~
../src/file.h:22:1: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
   21 | #include <string>
  +++ |+#include <cstdint>
   22 |
../src/file.h:60:5: error: ‘uint8_t’ does not name a type
   60 |     uint8_t read8();
      |     ^~~~~~~
../src/file.h:60:5: note: ‘uint8_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
../src/file.h:61:5: error: ‘uint16_t’ does not name a type
   61 |     uint16_t read16le();
      |     ^~~~~~~~
../src/file.h:61:5: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
../src/file.h:62:5: error: ‘uint32_t’ does not name a type
   62 |     uint32_t read32le();
      |     ^~~~~~~~
../src/file.h:62:5: note: ‘uint32_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
../src/file.h:63:5: error: ‘uint64_t’ does not name a type
   63 |     uint64_t read64le();
      |     ^~~~~~~~
../src/file.h:63:5: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
../src/file.h:64:5: error: ‘uint16_t’ does not name a type
   64 |     uint16_t read16be();
      |     ^~~~~~~~
../src/file.h:64:5: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
../src/file.h:65:5: error: ‘uint32_t’ does not name a type
   65 |     uint32_t read32be();
      |     ^~~~~~~~
../src/file.h:65:5: note: ‘uint32_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
../src/file.h:66:5: error: ‘uint64_t’ does not name a type
   66 |     uint64_t read64be();
      |     ^~~~~~~~
../src/file.h:66:5: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
../src/file.h:69:17: error: ‘uint8_t’ has not been declared
   69 |     void write8(uint8_t i);
      |                 ^~~~~~~
../src/file.h:70:20: error: ‘uint16_t’ has not been declared
   70 |     void write16le(uint16_t i);
      |                    ^~~~~~~~
../src/file.h:71:20: error: ‘uint32_t’ has not been declared
   71 |     void write32le(uint32_t i);
      |                    ^~~~~~~~
../src/file.h:72:20: error: ‘uint64_t’ has not been declared
   72 |     void write64le(uint64_t i);
      |                    ^~~~~~~~
../src/file.h:73:20: error: ‘uint16_t’ has not been declared
   73 |     void write16be(uint16_t i);
      |                    ^~~~~~~~
../src/file.h:74:20: error: ‘uint32_t’ has not been declared
   74 |     void write32be(uint32_t i);
      |                    ^~~~~~~~
../src/file.h:75:20: error: ‘uint64_t’ has not been declared
   75 |     void write64be(uint64_t);
      |                    ^~~~~~~~
../src/file.h:77:40: error: ‘uint64_t’ has not been declared
   77 |     void write(const unsigned char *s, uint64_t size);
      |                                        ^~~~~~~~
../src/file.h:78:31: error: ‘uint64_t’ has not been declared
   78 |     void write(const char *s, uint64_t size);
      |                               ^~~~~~~~

--- a/src/decompress.cpp
+++ b/src/decompress.cpp
@@ -53,6 +53,7 @@
 #include <cstring>
 
 #include <memory>
+#include <stdexcept>
 
 #define CHUNK 1024 * 1024
 

--- a/src/file.h
+++ b/src/file.h
@@ -19,6 +19,7 @@
 
 #include <cstdio>
 #include <string>
+#include <cstdint>
 
 class MMapper;
 

